README.md

```diff
@@ -29,56 +29,124 @@ The package helps collect device data for merchants with direct integration (sta
+        public enum RiskEnvironment {
+            case qa
+            case sandbox
+            case production
+        }
+        ```
+        </details>
+  4. Use the `configure` to complete your setup, then publish the device data within the closure with the `publishData` method. 
 
-    <details>
-    <summary>Arguments</summary>
+        <details>
+        <summary>Type definitions</summary>
 
-    ```swift
-    public struct RiskConfig {
-        public let publicKey: String
-        public let environment: RiskEnvironment
-        public let framesMode: Bool
-        
-        public init(publicKey: String, environment: RiskEnvironment, framesMode: Bool = false) {
-            self.publicKey = publicKey
-            self.environment = environment
-            self.framesMode = framesMode
+        ```swift
+        public struct PublishRiskData {
+            public let deviceSessionId: String
         }
-    }
 
-    public enum RiskEnvironment {
-        case qa
-        case sandbox
-        case prod
-    }
-    ```
-    </details>
+        public enum RiskError: LocalizedError, Equatable {
+            case configuration(Configuration)
+            case publish(Publish)
+        }
+
+        public extension RiskError {
+            enum Configuration: LocalizedError {
+                case integrationDisabled
+                case couldNotRetrieveConfiguration
+                
+                
+                public var errorDescription: String? {
+                    switch self {
+                    case .integrationDisabled:
+                        return "Integration disabled"
+                        
+                    case .couldNotRetrieveConfiguration:
+                        return "Error retrieving configuration"
+                    }
+                }
+            }
+            
+            enum Publish: LocalizedError {
+                case couldNotPublishRiskData
+                case couldNotPersisRiskData
+                case fingerprintServiceIsNotConfigured
+                
+                
+                public var errorDescription: String? {
+                    switch self {
+                    case .couldNotPublishRiskData:
+                        return "Error publishing risk data"
+                        
+                    case .couldNotPersisRiskData:
+                        return "Error persisting risk data"
+                        
+                    case .fingerprintServiceIsNotConfigured:
+                        return "Fingerprint service is not configured. Please call configure() method first."
+                    }
+                }
+            }
+            
+        }
+        ```
+        </details>
+
+See example below:
+```swift
+import Risk
+
+// Example usage of package
+let yourConfig = RiskConfig(publicKey: "pk_qa_xxx", environment: RiskEnvironment.qa)
 
+self.riskSDK = Risk.init(config: yourConfig)  
+
+self.riskSDK.configure { configurationResult in
```

Comment:
```suggestion
self.riskSDK.configure { [weak self] configurationResult in
    guard let self { return }
```

better to have a best practice typed down in the sample code, in case they are using a class they'd get a retain cycle otherwise

Reply:
That would result in an error like this: `'weak' may only be applied to class and class-bound protocol types, not 'ContentView'`

Reply:
they probably won't use the Risk SDK directly inside the view type. They'd rather put the business logic under another class or struct. If they choose to use a struct, that's fine. If they use a class and use our recommended method, they get a memory leak and the Risk SDK will never get deallocated. 

Considering that there is no reference of risk SDK method calls' requirement to be inside a value type data structure, then it's better to have this as a precaution. That's what I believe but I don't have strong opinions.